test/kcas/test.ml

```diff
@@ -564,6 +571,47 @@ let test_call () =
 
 (* *)
 
+(** This is a non-deterministic test that might fail occasionally. *)
+let test_timeout () =
+  Domain_local_timeout.set_system (module Thread) (module Unix);
+
+  let check (op : ?timeoutf:float -> bool Loc.t -> unit) () =
+    let x = Loc.make false in
+    let finally =
+      Domain_local_timeout.set_timeoutf 0.3 @@ fun () -> Loc.set x true
+    in
+    Fun.protect ~finally @@ fun () ->
+    (match op ~timeoutf:0.1 x with
+    | () -> assert false
+    | exception Timeout.Timeout -> ());
```

Comment:
I used `Alcotest.check_raises` here, but I saw the following error on CI

```
[exception] Stdlib.Queue.Empty
            Raised at Stdlib__Domain.join in file "domain.ml", line 258, characters 16-24
            Called from Stdlib__List.iter in file "list.ml", line 110, characters 12-15
            Called from Alcotest_engine__Core.Make.protect_test.(fun) in file "src/alcotest-engine/core.ml", line 181, characters 17-23
            Called from Alcotest_engine__Monad.Identity.catch in file "src/alcotest-engine/monad.ml", line 24, characters 31-35
```

which I assume comes from inside Alcotest, because `Stdlib.Queue` is not used anywhere in kcas.